src/coreclr/vm/jithelpers.cpp

```diff
@@ -5219,15 +5219,11 @@ void JIT_Patchpoint(int* counter, int ilOffset)
 // Unlike regular patchpoints, partial compilation patchpoints
 // must always transitio.
 //
-void JIT_PartialCompilationPatchpoint(int ilOffset)
+HCIMPL1(VOID, JIT_PartialCompilationPatchpoint, int ilOffset)
```

Comment:
Does `JIT_Patchpoint` need the same treatment?

Reply:
It doesn't loop, so (at least so far) I've never seen it cause problems. 

If it does end up jitting a method, that thread switches to preemptive, and any other threads that hit the patchpoint don't need to wait, they just keep running the Tier0 code.

Reply:
Does the `HELPER_METHOD_FRAME_BEGIN_0` setup need to be moved in front of `EECodeInfo codeInfo(ip);` for the same reason you had to move here?